src/game/objects/King/KingIdle.js

```diff
@@ -0,0 +1,69 @@
+import { AssetsManager } from '../../assets';
+import { Vector } from '../../shared';
+import { KingAnimated } from './KingAnimated';
+
+const SPRITE_ANIMATION_UPDATE = 10;
+
+class KingIdle extends KingAnimated {
+  /**
+   * @param {import('./King').King} king
+   */
+  constructor(king) {
+    super(
+      king,
+      [
+        AssetsManager.kingIdle01,
+        AssetsManager.kingIdle02,
+        AssetsManager.kingIdle03,
+        AssetsManager.kingIdle04,
+        AssetsManager.kingIdle05,
+        AssetsManager.kingIdle06,
+        AssetsManager.kingIdle07,
+        AssetsManager.kingIdle08,
+        AssetsManager.kingIdle09,
+        AssetsManager.kingIdle10,
+        AssetsManager.kingIdle11,
+      ],
+      SPRITE_ANIMATION_UPDATE,
+    );
+    // eslint-disable-next-line no-param-reassign
+    king.rigidbody.velocity = Vector.Zero;
+  }
+
+  /**
+   * @param {import('../../shared').Frame} frame
+   */
+  update(frame) {
+    if (this.king.keyboard.pressed('ArrowLeft')) {
+      return this.king.transitionState('runLeft').onUpdate(frame);
+    }
+    if (this.king.keyboard.pressed('ArrowRight')) {
+      return this.king.transitionState('runRight').onUpdate(frame);
+    }
+    if (this.king.keyboard.pressed('x')) {
+      if (this.king.canAttack) {
+        this.king.delayAttack();
+        return this.king.transitionState('attack').onUpdate(frame);
+      }
+      return super.update(frame);
+    }
+    if (this.king.keyboard.pressed('ArrowUp')) {
+      return this.king.transitionState('jump').onUpdate(frame);
+    }
+    if (this.king.keyboard.pressed('ArrowDown')) {
+      return this.king.transitionState('doorIn').onUpdate(frame);
+    }
+    if (this.king.rigidbody.velocity.y > 1) {
```

Comment:
To nie jest idealne rozwiązanie, bo nie wykrywa powolnego spadania, ale w sumie to raczej najprostszy sposób na uniknięcie dziwnych bugów z malutkimi prędkościami także ok.